main.rb

```diff
@@ -0,0 +1,122 @@
+# Enumerable
+module Enumerable
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    if is_a?(Array)
+      size.times do |i|
+        yield self[i]
+      end
+    else size.times do |i|
+      yield [keys[i], values[i]]
+    end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    size.times do |i|
+      yield self[i], i
+    end
+  end
+
+  def my_select
+    return to_enum(__method__) unless block_given?
+
+    array = []
+    my_each do |i|
+      array << i if yield i
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if block_given?
+      my_each { |i| return false unless yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return false unless i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return false if (args[0] =~ i).nil? }
+    else
+      my_each { |i| return false unless i.nil? }
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if block_given?
+      my_each { |i| return true if yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return true if i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return true unless (args[0] =~ i).nil? }
+    else
+      my_each { |i| return true if i.nil? }
+    end
+    false
+  end
+
+  def my_none?(*args)
+    if block_given?
+      my_each { |i| return false if yield i }
+    elsif args != [] && args[0].class != Regexp
+      my_each { |i| return false if i.is_a?(args[0]) }
+    elsif args[0].class == Regexp
+      my_each { |i| return false unless (args[0] =~ i).nil? }
+    else
+      my_each { |i| return false if i == true }
+    end
+    true
+  end
+
+  def my_count(*args)
+    @var = self
+    empty_arr = []
+
+    if block_given?
+      my_each { |i| empty_arr << i if yield i }
+
+      empty_arr.length
+    else
+      !args.empty? ? @var.length - (self - args).length : length
+    end
+  end
+
+  def my_map(*args)
+    return to_enum(__method__) unless block_given? || args[0].class == Proc
+
+    array = []
+    args[0].class == Proc ? my_each { |i| array << args[0].call(i) } : my_each { |i| array << (yield i) }
+    array
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] returns a new array with the results of running a given block once for every element in enum::range.
```ruby
(0..10).my_map {|num| num < 10 } == (0..10).map {|num| num < 10 
```
- Yours is throwing an error
![image](https://user-images.githubusercontent.com/52209067/85831734-f9d2e680-b74b-11ea-968c-cd87dc188258.png)

- Real all
![image](https://user-images.githubusercontent.com/52209067/85831785-0eaf7a00-b74c-11ea-8f3b-45eadcfdfbe1.png)